model/CHEM_DRV.F90

```diff
@@ -2248,20 +2277,88 @@ SUBROUTINE accumGCsubdd
        do k=1,subdd%ndiags
           ntm_loop: do n=1,State_Chm%nSpecies
              ! tracer 3D mixing ratios (SUBDD names are just tracer name):
-             if( trim( State_Chm%SpcData(N)%Info%Name ) .eq. trim(subdd%name(k)) ) then                
+             if( trim( State_Chm%SpcData(N)%Info%Name ) .eq. trim(subdd%name(k)) ) then
                 DO L=1,LmaxSUBDD
                 DO J=J_0,J_1
                 DO I=I_0,I_1
                    II = I - I_0 + 1
                    JJ = J - J_0 + 1
-                   sddarr3d(I,J,L) = State_Chm%Species(N)%Conc(II,JJ,L)         
+                   sddarr3d(I,J,L) = State_Chm%Species(N)%Conc(II,JJ,L)
                 ENDDO
                 ENDDO
                 ENDDO
+                call inc_subdd(subdd,k,sddarr3d)
+                exit ntm_loop
+             end if
+          end do ntm_loop
+       enddo ! k
+    enddo ! igroup
+
+    do igrp=1,ngroups
+       subdd => subdd_groups(grpids(igrp))
+       do k=1,subdd%ndiags
```

Comment:
I wasn't sure that we had the looping logic right here. I've separated it out for now but we should follow up in #99 once this is merged in.